cert/aziot-certd/src/lib.rs

```diff
@@ -106,15 +123,46 @@ impl Api {
             return Err(Error::Unauthorized(user, id));
         }
 
-        let x509 = create_cert(
-            &mut *this,
-            &id,
-            &csr,
-            issuer
-                .as_ref()
-                .map(|(issuer_cert, issuer_private_key)| (&**issuer_cert, issuer_private_key)),
-        )
-        .await?;
+        let req = X509Req::from_pem(&csr)
+            .or_else(|_| -> Result<_, BoxedError> {
+                let bytes = base64::decode(&csr)?;
+                Ok(X509Req::from_der(&bytes)?)
+            })
+            .map_err(|err| Error::invalid_parameter("csr", err))?;
+        let pubkey = req.public_key()
+            .map_err(|err| Error::invalid_parameter("csr", err))?;
+
+        if !req.verify(&pubkey)
+            .map_err(|err| Error::invalid_parameter("csr", err))?
+        {
+            Err(Error::invalid_parameter(
+                "csr",
+                "CSR failed to be verified with its public key".to_owned()
+            ))?
```

Comment:
Use `return Err();` instead of `Err()?`

(I'm surprised https://rust-lang.github.io/rust-clippy/master/index.html#try_err didn't fire. `clippy::all` should've enabled it.)

Reply:
Fixed in e6688907b6598cb9b4b2b38a4f3fd9ec86567e11.